drafts/tech-debt.md

```diff
@@ -0,0 +1,290 @@
+
+## **Making the Investment Work: Practical Guidance**
+
+Here’s how to embed this mindset and practice in your organization:
+
+### **📌 Track It Transparently**
+
+Use labels like `tech-debt` in your backlog. Make it visible to product owners and leadership.
+Frame it as risk mitigation and long-term acceleration — not “cleanup.”
+However, be mindful that such items might be de-prioritized in favor of new features.
+
+### **🖇️ Tag Refactorings to Features**
+
+Associate necessary refactoring efforts directly with feature development.
+This ensures that addressing technical debt becomes part of the feature implementation process,
+improving the codebase while delivering new functionalities.
+
+### **📊 Measuring the Invisible**
+
+One challenge with technical debt is that it’s not obvious in Jira or burn-down charts.
+But it *can* be tracked with meaningful metrics:
+
+| Metric                          | What It Tells You                                          |
+|---------------------------------|------------------------------------------------------------|
+| **Code Complexity**             | Risk of bugs and change difficulty                         |
+| **Technical Debt Ratio**        | Effort spent on fixes vs. features                         |
+| **Defect Density**              | Quality of code and test coverage                          |
+| **Code Churn Rate**             | Stability of recently changed code                         |
+| **Time to First Commit (TTFC)** | Onboarding friction for new devs                           |
+| **Lead Time & MTTR**            | Operational impact of tech quality                         |
+| **Technical Debt Heatmap**      | Highlighting significant areas which have technical debt's |
+
+By integrating these strategies, teams can manage technical debt proactively,
+ensuring it aligns with ongoing development efforts and business objectives.
+
+> We recommend adding a `tech debt dashboard` to your engineering metrics, ideally surfaced in
+> retros
+> or quarterly planning.
+
+### **🛠 Include in Sprint Planning**
+
+Allocate 10% of each sprint for debt reduction. Don’t let it become a stretch goal.
+
+> ℹ️:
+> _Allocating a portion of each sprint to address technical debt is a widely recognized practice in agile development._ 
+> _This approach ensures that code quality is maintained alongside the delivery of new features._ 
+> _The specific percentage allocated can vary based on the team's needs and project requirements._
+
+### **📣 Share the Wins**
+
+Demo improved test coverage, removed duplication, or better CI pipelines. Celebrate small wins to
+show value.
+
+### **🤝 Involve Everyone**
+
+Let developers decide what to tackle — from removing dead code to automating painful deployments.
+
+### **⚖ Prioritize Strategically**
+
+Balance high-impact fixes with low-hanging fruit. Use metrics like:
+
+- Code complexity
+- Defect density
+- Code churn
```

Comment:
In stead of code churn, I would prioritise based on module stability. When we look at the revision history of a codebase, we often see 90% of the code changes focused to only 10% of the code modules.

So addressing code complexity in a code module that is likely to never change would have very low-impact.